src/cls/rgw/cls_rgw_types.h

```diff
@@ -91,8 +91,17 @@ struct rgw_bucket_pending_info {
 };
 WRITE_CLASS_ENCODER(rgw_bucket_pending_info)
 
+
+enum class RGWObjCategory : uint8_t {
+  None      = 0,
```

Comment:
could we annotate (//) with the meaning/intent of each category?

Reply:
Good idea; take a look and see if it's adequate. I did audit all uses of each of the 4 enums.